src/main/java/seedu/address/logic/commands/RemoveTagCommand.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_INPUT;
+import static seedu.address.commons.core.Messages.MESSAGE_TAG_NOT_FOUND;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.person.exceptions.PersonNotFoundException;
+import seedu.address.model.person.exceptions.TagNotFoundException;
+
+
+/**
+ * Remove a tag from the tag lists of the address book and all persons in the address book
+ */
+public class RemoveTagCommand extends UndoableCommand {
+    public static final String COMMAND_WORD = "remove";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ":Removes the tag identified by the user input from the tag lists of the address book and all"
+            + " persons in the address book.\n"
+            + "Parameters: TAG NAME (must be alphanumeric)\n"
+            + "Example: " + COMMAND_WORD + " friend";
+
+    public static final String MESSAGE_REMOVE_TAG_SUCCESS = "Removed Tag: %1$s";
+
+    private final String targetTag;
+
+    public RemoveTagCommand(String targetTag) {
+        this.targetTag = targetTag;
+    }
+
+
+    @Override
+    public CommandResult executeUndoableCommand() throws CommandException {
+        try {
+            model.removeTag(targetTag);
+        } catch (IllegalValueException ive) {
```

Comment:
If `removeTagCommandParser` accounts for illegal tag values, this should be an assertion rather than a new exception (if the parser checks the value before creating a `removeTagCommand`, it should be impossible for a `removeTagCommand` to have an illegal value).

Reply:
This is true; however in the RemoveTagCommandTest if I want to test if the RemoveTagCommand can handle the invalid input, there is no way for me to test for parser there, i.e. the RemoveTagCommand has to handle the exception itself but not an assertion...I think if there is another command like `assertFalse` that allows me to test for assertion then i will change this part to assertion; or else i just remove the test? or I write a removeTagCommandParserTest?

Reply:
It is okay to remove that particular test I think as `RemoveTagCommand` must have a valid tag as a precondition, so it should never be encountered anyway.

Reply:
okie then i will write a RemoveTagCommandParserTest to cover this